File: Q8/Q8.c
/*Sum of a given number*/
#include <stdio.h>
int sum(int num)
 {
    if (num == 0) 
    {
        return 0;
    }
    return (num % 10) + sum(num / 10); 
}

void main() 
{
    int num;

    printf("Enter a number: \n");
    scanf("%d", &num);

    printf("Sum of digits : %d\n", sum(num));

}

File: Q8/Q8.ts
// Return the sum of digits of a number.

function sumOfDigitsTS(num: number): number {
  let sum = 0;
  while (num > 0) {
    sum += num % 10;
    num = Math.floor(num / 10);
  }
  return sum;
}
console.log(sumOfDigitsTS(1234));
